### PR TITLE
Implement hash records

### PR DESCRIPTION
**Pull request by gaetschwartz**
# Description

This PR implements hash maps using lazy records. They can be created like so:
```nu
{shell: "Nushell", best: true, uwu: "owo"} | into hash-record
```

This uses Hashmaps combined with lazy record to allow *faster* lookup. Note that to obtain even better performance, a custom non-default hashing method is used ([`ahash`](https://crates.io/crates/ahash)).

WhatsApp Image 2023-10-17 at 14 14 59_57840df3

# User-Facing Changes

Adds a new command: `into hash-record`

## Additional notes
This PR also fixes an issue with tests on windows, on certain setups, `cargo` reports the binary name as `cargo.exe` which makes some tests that expect the stdout to contain `cargo check` to fail.

### CODE REVIEW

File: crates/nu-command/src/conversions/into/record_hashmap.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            span,
+        ),
+    };
+    Ok(map.into_pipeline_data())
+}
+
+#[derive(Clone)]
+pub struct NuHashMapRecord {
+    hash_map: HashMap<String, Value>,
+    cols: Vec<String>,
```
Comment (atahabaki) [inline crates/nu-command/src/conversions/into/record_hashmap.rs:85]: IMO `cols` is completely unnecessary. Can't we can use `hash_map.keys()`?

**Review by CAD97 — COMMENTED** [review]
(Whoops this got buffered and never submitted)

**Comment by devyn** [comment]
I think rather than to take this, we're more likely to use some variant of `IndexMap` for the `Record` type if we can get the performance on small records to be good enough. That way you don't need a special type.

@sholderbach thoughts? I know we were talking about this recently. Trying to take a look at some older PRs.

**Comment by sholderbach** [comment]
Yup, let's tune `Record` instead. (we may want to keep the smaller single Vec solution for small-N to save space and `IndexMap` for optmized large N access, after measuring the tradeoffs)

Having a separate type would require much more effort to make sure all commands and operations support both the linear-search record and the hash version. Getting `LazyRecord` into some semblance of consistent support took/still takes a village.

Thanks for trying and pointing out the deficiency of the non-hashing record @gaetschwartz

**Pull request closed without merging**